lib/spack/spack/fetch_strategy.py

```diff
@@ -449,7 +460,10 @@ def archive(self, destination):
         if not self.archive_file:
             raise NoArchiveFileError("Cannot call archive() before fetching.")
 
-        shutil.copyfile(self.archive_file, destination)
```

Comment:
@opadron I think the following is causing #13404:

* The archive file is determined in part by the basename of the URL, in this case `libuuid-1.0.3.tar.gz?r=http%3A%2F%2Fsourceforge.net%2Fprojects%2Flibuuid%2F&ts=1433881396&use_mirror=iweb`
* Originally the `shutil.copyfile` command happily used this as the source name of the file (which was weird, but it worked)
* Now the `web.push_to_url` function is used, and it modifies the supplied local path with `url_util.parse(local_path)`

I think that

* `web.push_to_url` should not format the input path (the local file is what it is) i.e. remove the `url_util.parse(local_path)`
* arguably the archive filename should be formatted better (but technically that is not required) which is to say that the archive name should be based on `url_util.parse` of the basename of the URL

Would just removing `url_util.parse(local_path)` cause problems that you know of?